### ISSUE TITLE
Rendering exception occurred

### ISSUE THREAD

**Issue by SHU17** [saltstack/salt#34674]
I upgraded salt master & minion to version -- salt 2016.3.1 (Boron).  Post upgrade I get following exception (It was working fine before the upgrade):

```
2016-07-14 17:08:06,903 [salt.utils.templates                                 ][ERROR   ][11288] Rendering exception occurred: Jinja variable 'dict' object has no attribute 'osmajorrelease'
2016-07-14 17:08:06,904 [salt.pillar                                          ][CRITICAL][11288] Rendering SLS 'ldap' failed, render error:
```

Here is Pillar sls:

```
{% if grains['osmajorrelease'] == '5' %}
  ldap_nscd: nss_ldap
{% elif grains['osmajorrelease'] == '6' %}
  ldap_nscd: sssd-ldap
{% else %}
  ldap_nscd: sssd-ldap
{% endif %} 
```

**Comment by uvsmtid** [comment]
So, maybe check if you grains contain `osmajorrelease` key:

```
salt-call grains.items | tee salt.output
grep osmajorrelease salt.output
```

Check documentation (read release notes) if this key is still available in `2016.3.1`. If it does not available, update your pillars files.

Also, try to keep issue title more specific (like adding "due to missing `osmajorrelease` key in grains").

**Comment by Ch3LL** [comment]
@SHU17 were you able to try @uvsmtid suggestion to see if osmajorrelease key is a grain on your minion by running that command above? Can you paste the output of that command?

Alos what OS is this on? I'm assuming centos 5/6 but any clarification would help to what minion you are running this against. Thanks
